### PR TITLE
Add support for fontFileUrl

### PR DESCRIPTION
**Pull request by beerecca**
Reviewers: @pietvanzoen

### CODE REVIEW

File: lib/button-svg.js [hunk 1]
```diff
@@ -27,9 +27,17 @@ class ButtonSVG {
...
 
   generateStyle() {
+    const {fontCssUrl, fontFileUrl, fontName} = this.options;
+    const style = (fontCssUrl) ?
+      `@import url(${fontCssUrl}); text{font-family:"${fontName}";}` :
+      `@font-face {
+        font-family: "${fontName}";
+        src: url('${fontFileUrl}');
+      }
+      text{font-family:"${fontName}";}`;
```
Comment (pietvanzoen) [inline lib/button-svg.js]: space between `text` and `{`? 😬

File: lib/button-svg.js [hunk 2]
```diff
@@ -27,9 +27,17 @@ class ButtonSVG {
   }
 
   generateStyle() {
+    const {fontCssUrl, fontFileUrl, fontName} = this.options;
+    const style = (fontCssUrl) ?
+      `@import url(${fontCssUrl}); text{font-family:"${fontName}";}` :
```
Comment (pietvanzoen) [inline lib/button-svg.js]: I think we might be safer using quotes around the url right?

File: lib/button-svg.js [hunk 3]
```diff
@@ -27,9 +27,17 @@ class ButtonSVG {
...
+      `@font-face {
+        font-family: "${fontName}";
+        src: url('${fontFileUrl}');
```
Comment (pietvanzoen) [inline lib/button-svg.js]: let's use consistent quotes... don't mind which really

File: lib/button-svg.js [hunk 4]
```diff
@@ -27,9 +27,17 @@ class ButtonSVG {
   }
 
   generateStyle() {
+    const {fontCssUrl, fontFileUrl, fontName} = this.options;
+    const style = (fontCssUrl) ?
```
Comment (pietvanzoen) [inline lib/button-svg.js]: could remove the `text` style duplication with something like this:
```js
let style = (fontCssUrl) ?
  `@import url("${fontCssUrl}");` :
  `@font-face {font-family: "${fontName}"; src: url('${fontFileUrl}');}`;

style += `text {font-family:"${fontName}";}`;
```

Comment (pietvanzoen) [inline lib/button-svg.js]: oh... i guess we don't want any font face setting if the url isn't available right?

**Comment by pietvanzoen** [comment]
Perhaps best to break out the `style` generation into it's own static function of some sort that we can test.

**Comment by beerecca** [comment]
@pietvanzoen should've said, have updated

**Pull request merged**